### ISSUE TITLE
Unable to run containers with docker's system dirs/sockets mounted (v4.11.0)

### ISSUE THREAD

**Issue by nolar** [docker/for-mac#6433]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior

This command should work:

```
$ docker run -it -v /var/lib/docker/containers:/blahblah ubuntu echo hello
````

### Actual behavior

```
$ docker run -it -v /var/lib/docker/containers:/blahblah ubuntu echo hello
docker: Error response from daemon: error while creating mount source path '/host_mnt/private/var/lib/docker/containers': mkdir /host_mnt/private/var/lib/docker: operation not permitted.
```

### Information

After upgrading to Docker Desktop 4.11.0, I am unable to run containers that require Docker's socket or system folders. It worked with 4.9.1.

Downgrading Docker Desktop to 4.10.1 (downloaded from the [Release Notes](https://docs.docker.com/desktop/release-notes/) page) does help:

```
$ docker run -it -v /var/lib/docker/containers:/blahblah ubuntu echo hello
hello
```

This is needed to run such things as e.g. DataDog Agent, which monitors the containers' logs and metrics (DataDog Agent is now unable to start because of this).

  - macOS Version: 12.5
  - Intel chip or Apple chip: Intel
  - Docker Desktop Version: 4.11.0

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

```
$ /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0031: does the Docker API work?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0001: is the application running?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0017: can a VM be started?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0007: is the backend responding?
[PASS] DD0014: are the backend processes running?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0012: is the VM networking working?
[PASS] DD0032: do Docker networks overlap with host IPs?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0019: is the com.docker.vmnetd process responding?
[PASS] DD0033: does the host have Internet access?
No fatal errors detected.
```

### Steps to reproduce the behavior

  1. Just a command line: `docker run -it -v /var/lib/docker/containers:/blahblah ubuntu echo hello`

**Comment by janerikmai** [comment]
same for
    -v /etc/machine-id:/etc-machine-id:ro \

**Comment by nicks** [comment]
thanks for reporting! I can repro this. Investigating...

**Comment by snocke** [comment]
Same here. Can't create a kind cluster sind update to Docker Desktop 4.11.0 (83626)

`ERROR: failed to create cluster: command "docker run --name stackable-worker --hostname stackable-worker --label io.x-k8s.kind.role=worker --privileged --security-opt seccomp=unconfined --security-opt apparmor=unconfined --tmpfs /tmp --tmpfs /run --volume /var --volume /lib/modules:/lib/modules:ro -e KIND_EXPERIMENTAL_CONTAINERD_SNAPSHOTTER --detach --tty --label io.x-k8s.kind.cluster=stackable --net kind --restart=on-failure:1 --init=false kindest/node:v1.24.0@sha256:0866296e693efe1fed79d5e6c7af8df71fc73ae45e3679af05342239cdc5bc8e" failed with error: exit status 125
Command Output: a87711b55296aa841efd3e98e14a6e7702488edbe9c79e7d16719aab14b478ab
docker: Error response from daemon: error while creating mount source path '/host_mnt/lib/modules': mkdir /host_mnt/lib: input/output error`

System Version:	macOS 12.5 (21G72)
Chip:	Apple M1 Pro

**Comment by leomao10** [comment]
Hey @nicks,

Do you know if it is only impacted macOS env or it could potentially impact other env like linux distro as well?

**Comment by p1-0tr** [comment]
@leomao10 - yes, the same problem is present in Docker Desktop for Linux

**Comment by leomao10** [comment]
Thanks for letting me know @p1-0tr, is there a generic issue that I can track the status?

**Comment by nicks** [comment]
This is fixed in the release notes for v4.11.1 https://docs.docker.com/desktop/release-notes/#docker-desktop-4111
